### DISCUSSION TITLE
[Thread usage]

### DISCUSSION THREAD

**Discussion by coskrrb2002** [idaholab/moose#24844] [Q&A General]
Hello moose users,

I am curious whether MOOSE can designate the typical thread to calculate.

For example:
Workstation with 128 threads (1 CPU)
64 threads for inputfile1.i
64 threads for inputfile2.i

If I run the input files with --n-threads=64 option, then only 64 threads do the calculations for the two input files.

So, the remaining 64 threads take a rest very well...

Is there any way to assign different threads to input files?

**Comment by hugary1995** [comment]
What happens if you fork two sessions (eg open two terminals) and use each to run an input file with 64 threads?

**Comment by GiudGiud** [comment]
Note that you can still use MPI on a threaded cpu. It will use more memory but it will scale better. 

For threading scalability you will want to test it on your machine and your problem to determine how many threads you can use on an input and still get a speed up
